File: benchmarks/corpus/go/sql-injection/source.go
package main

import (
	"database/sql"
	"fmt"
	"net/http"
)

func handleSearch(w http.ResponseWriter, r *http.Request, db *sql.DB) {
	username := r.URL.Query().Get("username")
	query := fmt.Sprintf("SELECT * FROM users WHERE username = '%s'", username)
	rows, err := db.Query(query)
	if err != nil {
		http.Error(w, "Database error", http.StatusInternalServerError)
		return
	}
	defer rows.Close()
	fmt.Fprintf(w, "Results found")
}

File: benchmarks/corpus/python/sql-injection/source.py
import sqlite3

def get_user(db_path, username):
    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE username = '{username}'"
    cursor.execute(query)
    results = cursor.fetchall()
    conn.close()
    return results

def login(db_path, username, password):
    users = get_user(db_path, username)
    for user in users:
        if user[2] == password:
            return True
    return False
